pkg/xds/client/helpers.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	corepb "github.com/cilium/proxy/go/envoy/config/core/v3"
+)
+
+func Node(projectNum int64, mesh string, locality *corepb.Locality, clientID, clientUserAgent string) *corepb.Node {
+	nodeId := fmt.Sprintf("projects/%d/networks/mesh:%s/nodes/%s", projectNum, mesh, clientID)
+	return &corepb.Node{
+		Id:            nodeId,
+		UserAgentName: "xDS client test",
```

Comment:
Is this code only used for testing?

Reply:
I removed cloud-provider-specific code from this pull request. Constructor takes `*node` directly